core/src/main/java/org/apache/carbondata/core/util/ValueCompressionUtil.java

```diff
@@ -138,8 +141,7 @@ public static CompressionFinder getCompressionFinder(Object maxValue, Object min
     // ''b' for decimal, 'l' for long, 'n' for double
     switch (measureStoreType) {
       case 'b':
-        return new CompressionFinder(COMPRESSION_TYPE.BIGDECIMAL, DataType.DATA_BYTE,
-            DataType.DATA_BYTE);
+        return getBigDecimalCompressorFinder(maxValue, minValue, 0, dataTypeSelected);
```

Comment:
if mantissa equals to 0 hear for big decimal? if it is, keep passing mantissa to getBigDecimalCompressorFinder